pkg/crypto.go

```diff
@@ -70,5 +80,36 @@ func (c Crypto) Setup(hmacFile string, masterFile string) (bool, error) {
 }
 
 func (c Crypto) Protect(value string) (string, error) {
-	return value, nil // TODO implement this
+	var form = map[string]any{"datum": value}
+	log.Infof("%s > Protecting text using crypto.", c.instance.ID())
+	requestFormData := c.instance.http.RequestFormData(form)
+	response, err := requestFormData.Post(EncryptJsonPath)
+
+	if err != nil {
+		log.Errorf("%s > cannot protect text using crypto: %s", c.instance.ID(), err)
+		return "", err
+	} else if response.IsError() {
+		log.Errorf("%s > cannot protect text using crypto: %s", c.instance.ID(), response.Status())
+		return "", nil
+	}
+
+	bodyReader := response.RawBody()
+
+	defer func(bodyReader io.ReadCloser) {
+		err := bodyReader.Close()
+		if err != nil {
+			log.Errorf("%s > cloud not close reader: %s", c.instance.ID(), err)
```

Comment:
please recall my thoughts regarding context sensitive logging;

imagine me being a qa and seeing in console error z"local_author > cloud not close reader"` - this gives me nothing informative

Reply:
hmm and IMO you overcomplicated reading response; please check other places in which I am unmarshalling response, for example: https://github.com/wttech/aemc/blob/main/pkg/osgi_event_manager.go#L26

Reply:
Thanks, this UnmarshalJSON error checker looks cleaner, but I have question, why are you not closing your reader? I think, It can generate leaks.

Reply:
nope; ReadCloser is closing automatically after reading; this is how it differs from just Reader interface ;)

Reply:
but checking closing IMO is redundant / is a tradeoff as 3rd party (web server) will not unintentionally fail well working program on our side

Reply:
hmmm I want to avoid repetitions of low-level code (boilerplate) and hide it in utility methods like `fmt.` so we should rather put closing inside such methods instead of doing what you are doing